src/components/notebook/NotebookContent.tsx

```diff
@@ -8,14 +8,24 @@ import { CellBetweener } from "./cell/CellBetweener.js";
 import { EmptyStateCellAdder } from "./EmptyStateCellAdder";
 import { contextSelectionMode$ } from "./signals/ai-context.js";
 import { focusedCellSignal$, hasManuallyFocused$ } from "./signals/focus.js";
+import { useHideAiCells } from "../../hooks/useHideAiCells";
 
 export const NotebookContent = () => {
   const { store } = useStore();
-  const cellReferences = useQuery(queries.cellsWithIndices$);
+  const allCellReferences = useQuery(queries.cellsWithIndices$);
+  const { hideAiCells } = useHideAiCells();
 
   const focusedCellId = useQuery(focusedCellSignal$);
   const hasManuallyFocused = useQuery(hasManuallyFocused$);
 
+  // Filter out AI cells if hideAiCells is true
+  const cellReferences = React.useMemo(() => {
+    if (!hideAiCells) {
+      return allCellReferences;
+    }
+    return allCellReferences.filter((cell) => cell.cellType !== "ai");
+  }, [allCellReferences, hideAiCells]);
```

Comment:
You could/should bake this into the query above or make it a whole new query so the filtering is done on the sqlite side rather than in JS.